app/controllers/reviews_controller.rb

```diff
@@ -32,7 +52,123 @@ def update
+    when "only_contacted"
+      render :new_only_contacted, status: :unprocessable_entity
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def new_review_attributes
+    @space = Space.find(params[:space_id]) unless defined? @review
+    @review = Review.new(space: @space) unless defined? @review
+    common_review_attributes
+  end
+
+  def common_review_attributes
+    @facility_reviews = @space.aggregated_facility_reviews
+  end
+
+  def parse_before_update(review_params, review)
+    review_params["facility_reviews_attributes"] = parse_facility_reviews(
+      review_params["facility_reviews_attributes"],
+      review
+    )
+    review_params
+  end
+
+  def parse_before_create(review_params)
+    review_params["facility_reviews_attributes"] = parse_facility_reviews(review_params["facility_reviews_attributes"])
+    review_params["user"] = current_user
+    review_params
+  end
+
+  def parse_facility_reviews(facility_reviews, review = nil)
+    # Converting to .values exists solely because I didn't manage to create a
+    # form with radio buttons that sent the facility_reviews in a way that could
+    # be automatically picked up by Rails AND still be seen as individual in the form itself.
+    # Thus, we have to parse what is sent from the form here into something
+    # that more closely resembles what the model expects:
+    # Changes welcome!
+    facility_review_values = facility_reviews.values
+
+    known_reviews = delete_all_unknown facility_review_values
+    add_user_and_space known_reviews, review
+  end
+
+  # Facility Reviews only not need a user or space attached, as
+  # they are attached to reviews, who have both of those already.
+  # However, refactoring that means changing a lot of how
+  # facility reviews are calculated. Feel free to submit a PR if you
+  # want to!
+  #
+  # In the meantime, this code makes sure that we have user and space
+  # set, even though the form doesn't send them for each facility review.
+  def add_user_and_space(facility_reviews, review = nil)
+    facility_reviews.map do |facility_review|
+      facility_review[:user] = review&.user ? review.user : current_user
+      facility_review[:space_id] = review&.space&.id ? review.space.id : review_params["space_id"]
+      facility_review
+    end
+  end
+
+  # This should be done in the model, but I never figured out how.
+  # I tried to .destroy and filter on before_validation, but
+  # it only got roll-backed.
+  def delete_all_unknown(facility_reviews)
+    facility_reviews.each do |facility_review|
+      destroy_if_unknown(facility_review)
+    end
+
+    facility_reviews.filter do |facility_review|
+      facility_review[:experience] != "unknown"
+    end
+  end
+
+  def destroy_if_unknown(facility_review)
+    return false unless facility_review[:experience] == "unknown"
+    return false if facility_review[:id].blank?
+
+    current = FacilityReview.find(facility_review[:id])
+    current&.destroy
+  end
```

Comment:
Tror dette kan bli gjort en del enklere med:
```rb
def delete_all_unknown
  ids = facility_reviews.filter_map { |review| review[:id] if review[:experience] == "unknown" }
  FacilityReview.where(id: ids).delete_all
end
```

Reply:
Den gjør to ting, som sikkert burde være tydeligere:

Før vi sender det videre til model, så gjør controller:

1) Sletter eventuelle allerede lagrede FacilityReviews som nå er satt til unknown

2) Ignorerer (ved å filtrere ut) facility_reviews som ikke allerede er lagret men har unknown

```ruby
def delete_all_unknown
  ids = facility_reviews.filter_map { |review| review[:id] if review[:experience] == "unknown" }
  FacilityReview.where(id: ids).delete_all
end
```

Vil kun hjelpe for den første, men sikkert ikke dumt uansett!

Tror vi også må teste for .id, eller hvordan håndterer delete_all det om den får masse `nil` i arrayen? facility_review[:id] sendes bare for de facility reviews som allerede finnes.

Reply:
filter_map vil ikke ta med `nil` values

Reply:
```rb
test = 
  [{id: nil, experience: "unknown"}, {id: 12, experience: "unknown"}, {id: 22, experience: "asda"}]
    .filter_map { |review| review[:id] if review[:experience] == "unknown" }
```
=>
`[12]`

Reply:
for å slette alle unknown fra facility_reviews arrayet kan du bruke
```rb
facility_reviews.delete_if {|review| review[:experience] == "unknown" }
```

Reply:
kult! Ser mye bedre ut! Fikser nå.

Reply:
Gjorde det med .destroy_all i stedet, for å sikre at aggregate facility reviews kjører - men endret nå. Takk for bedre kode!